### DISCUSSION TITLE
config: check that various theme options are specified

### DISCUSSION THREAD

**Discussion by jehoshua7** [McShelby/hugo-theme-relearn#67] [Q&A] [answered]
Trying to find out today how to either remove the "MORE" on the menu or change the wording.I can see the generated code in a number of .HTML files as `<h3>More</h3>`, which is obviously sourced from /i18n/en.toml , and the code for that is in /layouts/partial/menu.html , as `<h3>{{ if not $disableShortcutsTitle}}{{ T "Shortcuts-Title"}}{{ end }}</h3>`

I would expect this to be specified in the config.toml , yet it wasn't found.

A quick search of other 'disable' options used within the theme resulted in finding these:
* disableToc
* disableAssetsBusting
* disablesearch

meaning, these are used in the theme (plus disableShortcutsTitle) and not specified in config.toml

The search I did was quick and not thorough though, so there may be more. So, is a good idea to check that all the 'disable' options used in the theme are specified in the configuration ?  Or possibly it goes a little bit deeper, wherein there may exist a need to check that all optional components are setup in the configuration ??

**Comment by McShelby** [comment]
The `config.toml` is _not_ meant to be a documentation for all available config options. If you are looking for this the [Configuration](https://relearn.netlify.app/basics/configuration/) topic provides the necessary information.

Even more documentation for the `disableShortcutsTitle` is provided by the [Menu extra shortcuts](https://relearn.netlify.app/cont/menushortcuts/) topic.

**Comment by jehoshua7** [comment]
@McShelby - Thanks. In modifying the content I had removed those links you have mentioned. No wonder I couldn't find any reference to them.  :+1:

**Comment by jehoshua7** [comment]
It makes sense to have **all** the possible configuration parameters in `config.toml` as shown in [Global Site Paramers](https://relearn.netlify.app/basics/configuration/)

**Reply by McShelby** [reply]
> It makes sense to have **all** the possible configuration parameters in `config.toml` as shown in [Global Site Paramers](https://relearn.netlify.app/basics/configuration/)

No, because this `config.toml` has an actual use: It's used for the netlify deploy and therefore differs from default parameter values.

**Comment by jehoshua7** [comment]
> No, because this config.toml has an actual use: It's used for the netlify deploy and therefore differs from default parameter values.

Does that impact upon a deploy that **isn't** for Netify ?

**Reply by McShelby** [reply]
Some background here: The exampleSite of a theme repo has the following use cases:

- it provides a test bed for the developer
- it provides an **example** configuration for users of the theme; it is meant to be edited by the user as to his likings
- it serves as a documentation site for the theme; this is optional and the case for the relearn theme; therefore, the configuration serves a certain purpose and can not be changed arbitrarily

See also https://discourse.gohugo.io/t/how-do-i-do-theme-development-with-an-example-site-in-the-repo/2136/3

**Reply by jehoshua7** [reply]
@McShelby Thanks :+1:

**Comment by McShelby** [comment, marked as answer]
Taken from above as final answer (comments could not be marked as answer)

> It makes sense to have **all** the possible configuration parameters in `config.toml` as shown in [Global Site Paramers](https://relearn.netlify.app/basics/configuration/)
 
No, because this `config.toml` has an actual use: It's used for the netlify deploy and therefore differs from default parameter values.
